tests/mr/device/hwdecompress_tests.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <rmm/detail/error.hpp>
+#include <rmm/mr/device/cuda_async_memory_resource.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+
+#include <cuda.h>
+#include <cuda_runtime_api.h>
+
+#include <gtest/gtest.h>
+
+namespace rmm::test {
+namespace {
+
+class HWDecompressTest : public ::testing::Test {
+ protected:
+  static void check_decompress_capable(void* ptr)
+  {
+    int driver_version{};
+    RMM_CUDA_TRY(cudaDriverGetVersion(&driver_version));
+    auto const min_hw_decompress_version{12080};
+    if (driver_version >= min_hw_decompress_version) {
```

Comment:
TODO: I want to restructure this so that we have a clear skip message in the tests that says one of:
- `GTEST_SKIP() << "Tests were compiled with CUDA < 12.8";`
- `GTEST_SKIP() << "Tests are being run with CUDA < 12.8";`
depending on the circumstances.